presto-main/src/main/java/com/facebook/presto/sql/gen/PageProcessorCompiler.java

```diff
@@ -741,11 +743,15 @@ private static BytecodeBlock getBytecodeFilterOnDictionary(
         TryExpressionExtractor tryExtractor = new TryExpressionExtractor();
         projection.accept(tryExtractor, null);
         List<CallExpression> tryExpressions = tryExtractor.getTryExpressionsPreOrder();
+        Set<CallExpression> genTryExpressions = new HashSet<>();
```

Comment:
Instead of having a list and a set to track the same thing, could you instead modify `getTryExpressionsPreOrder`? You would need to ensure the ordering is correct, though.